.github/workflows/docker.yml

```
@@ -31,29 +31,41 @@ jobs:
       - name: Fetch origin/master from Git
         run: git fetch origin master
 
-      - name: Build Docker image
+      - name: Determine build attributes
+        id: attributes
         run: |
-          brew_version="$(git describe --tags --dirty --abbrev=7)"
-          echo "Building for Homebrew ${brew_version}"
-          docker build -t brew \
-               --build-arg=version=${{matrix.version}} \
-               --label org.opencontainers.image.created="$(date --rfc-3339=seconds --utc)" \
-               --label org.opencontainers.image.url="https://brew.sh" \
-               --label org.opencontainers.image.documentation="https://docs.brew.sh" \
-               --label org.opencontainers.image.source="https://github.com/${GITHUB_REPOSITORY}" \
-               --label org.opencontainers.image.version="${brew_version}" \
-               --label org.opencontainers.image.revision="${GITHUB_SHA}" \
-               --label org.opencontainers.image.vendor="${GITHUB_REPOSITORY_OWNER}" \
-               --label org.opencontainers.image.licenses="BSD-2-Clause" \
-               .
+          {
+            echo "date=$(date --rfc-3339=seconds --utc)"
+            echo "brew_version=$(git describe --tags --dirty --abbrev=7)"
+          } | tee -a "${GITHUB_OUTPUT}"
+
+      - name: Build Docker image
+        uses: docker/build-push-action@5cd11c3a4ced054e52742c5fd54dca954e0edd85 # v6
+        with:
+          push: false
```

```suggestion
          push: ${{ github.event_name == 'release' ||
          (github.event_name == 'push' && github.ref == 'refs/heads/master')
          (github.event_name == 'push' && github.ref == 'refs/heads/master' &&
           matrix.version == '22.04') }}
```

or similar so we don't need separate push steps later.

Would require login before so I've whitelisted [docker/login-action](https://github.com/docker/login-action) in case that makes things easier

---

A separate push step is probably still needed because we test the image before pushing. But we can do something like https://docs.docker.com/build/ci/github-actions/test-before-push/.